lib/Config/ServicesChecker.php

```diff
@@ -116,4 +117,17 @@ function isPremiumKeyValid($display_error_notice = true) {
 
     return false;
   }
+
+  function isFromEmailAuthorized($display_error_notice = true) {
+    $mta_log_error = $this->settings->get('mta_log.error', []);
+
+    if ($mta_log_error && $mta_log_error['operation'] === MailerError::OPERATION_AUTHORIZATION) {
+      if ($display_error_notice) {
+        WPNotice::displayError($mta_log_error['error_message'], 'js-error-unauthorized-email', '', false, false);
```

Comment:
I know it was done even before, but wouldn't be better to pass some flag to frontend instead of displaying a message from the database?

Reply:
Probably yes? I decided on this solution because it seems an easier solution since WP Notice is doing just that - showing a message on every page. And I wanted to do more PHP than JS stuff.